core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -663,6 +673,37 @@ public boolean canDeleteIndexContents(Index index, IndexSettings indexSettings)
     }
 
     /**
+     * Verify that the contents on disk for the given index is deleted; if not, delete the contents.
+     * This method assumes that an index is already deleted in the cluster state and/or explicitly
+     * through index tombstones.
+     * @param index {@code Index} to make sure its deleted from disk
+     * @param clusterState {@code ClusterState} to ensure the index is not part of it
+     * @return IndexMetaData for the index loaded from disk
+     */
+    @Nullable
+    public IndexMetaData verifyIndexIsDeleted(final Index index, final ClusterState clusterState) {
+        // this method should only be called when we know the index is not part of the cluster state
+        assert clusterState.metaData().hasIndex(index.getName()) == false;
```

Comment:
It's a shame that we have to pass the cluster state into this just to add an assert, how about making this a real check that blows up?